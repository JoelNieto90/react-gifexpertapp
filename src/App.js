import React, { Fragment, useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const App = () => {
	const [categories, setCategories] = useState(["Dragon Ball Z"]);

	return (
		<Fragment>
			<h2>GifExpertApp</h2>
			<AddCategory setCategories={setCategories} />
			<hr />
			<ol>
				{categories.map((category) => (
					<GifGrid key={category} category={category} />
				))}
			</ol>
		</Fragment>
	);
}
