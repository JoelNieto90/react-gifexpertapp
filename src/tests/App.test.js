import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { App } from "../App";

describe("Prueba <App />", () => {
	test("Debe de mostrar el componente <App />", () => {
		const wrapper = shallow(<App />);
		expect(wrapper).toMatchSnapshot();
	});
});
