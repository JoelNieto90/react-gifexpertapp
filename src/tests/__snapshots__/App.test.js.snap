// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <App /> Debe de mostrar el componente <App /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;
